src/lib/validator.ts

```diff
@@ -84,6 +89,52 @@ class Validator {
       errorMessage(`No '${key}' attribute, which is required.`);
     }
   }
+
+  /**
+   * If a username is included in the model attributes, then validate that
+   * the model was created by the owner of the username. This is done by matching
+   * the Gaia URL to any Gaia URL in that user's profile.json
+   */
+  async validateUsername(): Promise<boolean> {
+    if (!(this.attrs.username && this.gaiaURL)) {
```

Comment:
When is it possible that the gaiaURL is not set? If it is not set should then username be null so that it is not possible to create a model with a username that is not verified?

Reply:
The `gaiaURL` is set always on the client side if `validateUsername` is true. This should be updated to also validate for the presence of `gaiaURL`. Good catch, I'll add to the "todos"